### PR TITLE
Using recommended metrics in train_molnet.py

### PR DESCRIPTION
**Pull request by mottodora**

### CODE REVIEW

**Comment by mottodora** [comment]
Now there are some problems in this PR.
1. we can't save models because this implementation use defined function in 
`AttributeError: Can't pickle local object 'main.<locals>.loss_fun'`
https://github.com/mottodora/chainer-chemistry/blob/f35b10ad6e8bd19dba0cc1eb1d1338f51fefcdcf/examples/molnet/train_molnet.py#L179-L180

2. In some datasets, it seems Evaluator raises ValueError Because

**Comment by corochann** [comment]
1. as written in the error message
> AttributeError: Can't pickle local object 'main.<locals>.loss_fun'

you can try defining function in *global scope* and try pickle again?

2. 
> ('Only one class present in y_true. ROC AUC score is not defined in that case.',)
It means that dataset contains *only one of positive or negative label*. This is caused by may be too imbalanced dataset, but is it really expected behavior??

**Comment by corochann** [comment]
could you resolve conflict?

**Comment by mottodora** [comment]
Update.

File: examples/molnet/train_molnet.py [hunk 1]
```diff
@@ -189,20 +203,24 @@ def get_dataset_paths(cache_dir, num_data):
...
-        # print_report_targets.append('train/main/roc_auc')
-        trainer.extend(ROCAUCEvaluator(
-            val_iter, model, device=args.gpu, eval_func=predictor,
-            converter=concat_mols, name='val', raise_value_error=False))
-        print_report_targets.append('val/main/roc_auc')
+    for metric_name, metric_fun in metrics.items():
+        if isinstance(metric_fun, types.FunctionType):
+            print_report_targets.append('main/' + metric_name)
+            print_report_targets.append('validation/main/' + metric_name)
+        elif BatchEvaluator in inspect.getmro(metric_fun):
```
Comment (corochann) [inline examples/molnet/train_molnet.py]: what is the difference from `isinstance(metric_fun, BatchEvaluator)`?

File: examples/molnet/train_molnet.py [hunk 2]
```diff
@@ -189,20 +203,24 @@ def get_dataset_paths(cache_dir, num_data):
...
+            #     converter=concat_mols, name='train',
+            #     raise_value_error=False))
+            # print_report_targets.append('train/main/roc_auc')
+            trainer.extend(metric_fun(
+                val_iter, model, device=args.gpu, eval_func=predictor,
+                converter=concat_mols, name='val',
+                raise_value_error=False))
+            print_report_targets.append('val/main/' + metric_name)
+        else:
+            raise TypeError
```
Comment (corochann) [inline examples/molnet/train_molnet.py]: Could you add some message?

**Pull request merged**